src/extensions/extensionsApi.ts

```diff
@@ -430,14 +437,20 @@ export async function updateInstance(args: {
     body.config.params = args.params;
     updateMask += ",config.params";
   }
-  if (args.allowedEventTypes !== undefined) {
-    body.config.allowedEventTypes = args.allowedEventTypes;
-    updateMask += ",config.allowed_event_types";
-  }
 
-  if (args.eventarcChannel !== undefined) {
+  if (!args.canEmitEvents) {
+    body.config.allowedEventTypes = undefined;
+    body.config.eventarcChannel = undefined;
+    updateMask += ",config.allowed_event_types,config.eventarc_channel";
+  } else {
+    if (args.allowedEventTypes === undefined || args.eventarcChannel === undefined) {
+      throw new FirebaseError(
+        `This instance is configured to emit events, but either allowed event types or eventarc channel is undefined.`
+      );
+    }
+    body.config.allowedEventTypes = args.allowedEventTypes;
     body.config.eventarcChannel = args.eventarcChannel;
-    updateMask += ",config.eventarc_channel";
+    updateMask += ",config.allowed_event_types,config.eventarc_channel";
```

Comment:
You have this same line in both parts of an if else  statement - you can just move it up a level.

In fact, i think you can simplify this whole logic quite a bit to:
```
if (args.canEmitEvents) {
 if (args.allowedEventTypes === undefined || args.eventarcChannel === undefined) {
      throw new FirebaseError(
        `This instance is configured to emit events, but either allowed event types or eventarc channel is undefined.`
      );
    }
    body.config.allowedEventTypes = args.allowedEventTypes;
    body.config.eventarcChannel = args.eventarcChannel;
}
updateMask += ",config.allowed_event_types,config.eventarc_channel"; // or just include this when you define update mask.
```

Reply:
Done